controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -149,6 +149,11 @@ func (r *EphemeralRunnerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
+	if ephemeralRunner.Status.RunnerId == 0 {
```

Comment:
Hey, I don't think that changing the registration and then patching the finalizers should be the right way to go. However, I noticed that the order in which we applied finalizers is incorrect.
We first need to apply the `ephemeralRunnerFinalizerName`, then the `ephemeralRunnerActionsFinalizerName`.

Would you be interested in trying out that approach?

Reply:
Sure, will do the tests, but shouldn't we return Result also after ephemeralrunner.actions.github.com/runner-registration-finalizer apply?

Reply:
Good catch, we should!

Reply:
Looks good to me. I'll wait for your feedback on Monday before merging it.

Reply:
Can you please also fix the test :relaxed: The order of finalizers is different now

Reply:
> Looks good to me. I'll wait for your feedback on Monday before merging it.

We didn't observe the issue since Wednesday :)